mailpoet/assets/js/src/automation/listing/store/reducer.ts

```diff
@@ -0,0 +1,52 @@
+import { Action } from '@wordpress/data';
+import { State } from './types';
+import { Workflow } from '../workflow';
+
+export function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case 'ADD_NOTICE':
+      return {
+        ...state,
+        notices: [...state.notices, action.notice],
+      };
+    case 'SET_WORKFLOWS':
+      return {
+        ...state,
+        workflows: action.workflows,
+        workflowCount: action.workflows.length,
+      };
+    case 'SET_WORKFLOW_COUNT':
+      return {
+        ...state,
+        workflowCount: action.count,
+      };
+    case 'ADD_WORKFLOW':
+      return {
+        ...state,
+        workflows: [action.workflow, ...state.workflows],
+        workflowCount: [action.workflow, ...state.workflows].length,
+      };
+    case 'UPDATE_WORKFLOW':
+      return {
+        ...state,
+        workflows: state.workflows.map((workflow: Workflow) => {
+          if (workflow.id === action.workflow.id) {
+            return action.workflow;
+          }
+          return workflow;
+        }),
+      };
+    case 'REMOVE_WORKFLOW':
+      return {
+        ...state,
+        workflows: state.workflows.filter(
+          (workflow: Workflow) => workflow.id !== action.workflow.id,
+        ),
+        workflowCount: state.workflows.filter(
+          (workflow: Workflow) => workflow.id !== action.workflow.id,
+        ).length,
```

Comment:
Is there a better way to handle this, like having REMOVE_WORKFLOW somehow trigger SET_WORKFLOW_COUNT? I'm repeating myself here because there was an eslint warning about declaring variables inside a case statement.